lib/plugins/aws/provider.js

```diff
@@ -1806,6 +1846,17 @@ class AwsProvider {
       );
     }
   }
+
+  reportDeprecatedUsage(deprecationCode, pairs) {
```

Comment:
Let me know if I should test this function, and how :) Making it standalone would make it easier.

Reply:
We can put it into `lib/utils/report-deprecated-properties.js`

I'd also use object or map as an input (not array). It may work that way, and under the hood use [`utils/logDeprecation.js`](https://github.com/serverless/serverless/blob/816394c6e5dfc50b332314aef66eeb9ed75d139a/lib/utils/logDeprecation.js)

```javascript
reportDeprecatedProperties('PROVIDER_IAM_SETTINGS', {
  role: 'iam.role',
  ...
}, { serviceConfig: this.serverless.service });
```

Reply:
I can see file name casing is all over the place, should I use kebab case or camel case?

Reply:
We plan to rename all to kebab case. So you can introduce new util in kebab case